xfrd-tcp.c

```diff
@@ -639,6 +772,80 @@ tcp_conn_ready_for_reading(struct xfrd_tcp* tcp)
 	buffer_clear(tcp->packet);
 }
 
+#ifdef HAVE_TLS_1_3
+static int
+conn_write_ssl(struct xfrd_tcp* tcp, SSL* ssl)
+{
+	ssize_t sent;
+
+	if(tcp->total_bytes < sizeof(tcp->msglen)) {
+		uint16_t sendlen = htons(tcp->msglen);
+		// send
+		int request_length = sizeof(tcp->msglen) - tcp->total_bytes;
+		sent = SSL_write(ssl, (const char*)&sendlen + tcp->total_bytes,
+						 request_length);
+		switch(SSL_get_error(ssl,sent)) {
+			case SSL_ERROR_NONE:
+				if(request_length != sent)
+					log_msg(LOG_ERR, "xfrd: incomplete write of tls");
```

Comment:
It is normal to have incomplete writes because of the packetizing that happens, so it should not be a priority type LOG_ERR to print it during normal operations.  Perhaps debug or high verbosity printout, but it is not an operational issue.

Reply:
Resolved in current code.